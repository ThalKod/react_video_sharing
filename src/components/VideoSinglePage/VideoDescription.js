import  React from "react";

export default function VideoDescription({ description, tags}){
  return (
      <div className="info">
        <h4>Categories:</h4>
        <p>None Yet :)</p>

        <h4>About:</h4>
        <p>{description}</p>

        <h4>Tags:</h4>
        <p className="sv-tags">
          {tags.map(tag => {
            return <span key={tag.id}>{tag.text}</span>
          })}
        </p>
        <div className="row date-lic">
          <div className="col-lg-6">
            <h4>Release Date: </h4><br />
            <p>2 Days ago</p>
          </div>
        </div>
        <div className="showless">
          <a href="/">showless</a>
        </div>
      </div>
  )
};

