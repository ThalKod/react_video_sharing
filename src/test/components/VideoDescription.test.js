import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import VideoDescription from "../../components/VideoSingle/VideoDescription";

describe("VideoDescripition", () => {

  let wrapper;
  beforeEach( () => {
    const props = {
      description: "description",
      tags: [{id: "text", text: "text"}, {id: "hello", text: "hello"}],
      createdAt: 12992383299139
    };

    wrapper = shallow(<VideoDescription {...props} />);
  });

  it("should render the component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should show the full description on button click", () => {
    wrapper.find("button").simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.state("showDescription")).toBe(true);
  });
});
