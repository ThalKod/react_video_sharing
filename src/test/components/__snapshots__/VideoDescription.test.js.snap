// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoDescripition should render the component correctly 1`] = `
<div
  className="info"
>
  <h4>
    Categories:
  </h4>
  <p>
    None Yet :)
  </p>
  <CSSTransitionGroup
    transitionAppear={false}
    transitionEnter={true}
    transitionEnterTimeout={1000}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="menu"
  />
  <div
    className="showless"
  >
    <button
      onClick={[Function]}
      type="submit"
    >
      Show more
    </button>
  </div>
</div>
`;

exports[`VideoDescripition should show the full description on button click 1`] = `
<div
  className="info"
>
  <h4>
    Categories:
  </h4>
  <p>
    None Yet :)
  </p>
  <CSSTransitionGroup
    transitionAppear={false}
    transitionEnter={true}
    transitionEnterTimeout={1000}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="menu"
  >
    <div>
      <h4>
        About:
      </h4>
      <p>
        description
      </p>
      <h4>
        Tags:
      </h4>
      <p
        className="sv-tags"
      >
        <span
          key="text"
        >
          text
        </span>
        <span
          key="hello"
        >
          hello
        </span>
      </p>
      <div
        className="row date-lic"
      >
        <div
          className="col-lg-6"
        >
          <h4>
            Release Date: 
          </h4>
          <br />
          <p>
            in 362 years
          </p>
        </div>
      </div>
    </div>
  </CSSTransitionGroup>
  <div
    className="showless"
  >
    <button
      onClick={[Function]}
      type="submit"
    >
      Show Less
    </button>
  </div>
</div>
`;
